iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/AbstractCreateConnectableFeederBays.java

```diff
@@ -112,46 +169,68 @@ private boolean checkOrders(int side, VoltageLevel voltageLevel, ReportNode repo
-        if (positionRangeForSection.isEmpty()) {
-            LOGGER.warn("Positions of adjacent busbar sections do not leave slots for new positions on busbar section '{}'. No position extension created.", busbarSection.getId());
-            positionNoSlotLeftByAdjacentBbsReport(reportNode, busbarSection.getId());
-            return false;
+        Optional<Range<Integer>> rangeOpt = getPositionRange(busbarSection);
+        if (rangeOpt.isEmpty()) {
+            String msg = "Positions of adjacent busbar sections do not leave slots for new positions on busbar section '" + busbarSection.getId() + "'.";
+            return logAndReport(logOrThrowIfIncorrectPositionOrder, throwException, msg,
+                severity -> positionNoSlotLeftByAdjacentBbsReport(reportNode, busbarSection.getId(), severity)
+            );
         }
 
-        int minValue = positionRangeForSection.get().getMinimum();
-        if (positionOrder < minValue) {
-            LOGGER.warn("PositionOrder {} too low (<{}). No position extension created.", positionOrder, minValue);
-            positionOrderTooLowReport(reportNode, minValue, positionOrder);
-            return false;
+        Range<Integer> range = rangeOpt.get();
+        if (positionOrder < range.getMinimum()) {
+            String msg = POSITION_ORDER_STRING + positionOrder + " too low (<" + range.getMinimum() + ").";
+            return logAndReport(logOrThrowIfIncorrectPositionOrder, throwException, msg,
+                severity -> positionOrderTooLowReport(reportNode, range.getMinimum(), positionOrder, severity)
+            );
         }
 
-        int maxValue = positionRangeForSection.get().getMaximum();
-        if (positionOrder > maxValue) {
-            LOGGER.warn("PositionOrder {} too high (>{}). No position extension created.", positionOrder, maxValue);
-            positionOrderTooHighReport(reportNode, maxValue, positionOrder);
-            return false;
+        if (positionOrder > range.getMaximum()) {
+            String msg = POSITION_ORDER_STRING + positionOrder + " too high (>" + range.getMaximum() + ").";
+            return logAndReport(logOrThrowIfIncorrectPositionOrder, throwException, msg,
+                severity -> positionOrderTooHighReport(reportNode, range.getMaximum(), positionOrder, severity)
+            );
         }
-
         return true;
     }
 
+    private boolean logAndReport(boolean logOrThrowIfIncorrectPositionOrder, boolean throwException, String message,
+                                 Consumer<TypedValue> report) {
+        TypedValue severity = logOrThrowIfIncorrectPositionOrder ? TypedValue.ERROR_SEVERITY : TypedValue.WARN_SEVERITY;
+        if (logOrThrowIfIncorrectPositionOrder) {
+            LOGGER.error(message);
+        } else {
+            LOGGER.warn(message);
+        }
+
+        report.accept(severity);
+
+        if (logOrThrowIfIncorrectPositionOrder && throwException) {
+            throw new PowsyblException(message);
+        }
```

Comment:
Should this message still be logged if the exception is thrown right after?
I'm not really sure how you want to manage the log/report level here.
Would this proposal be correct?
```suggestion
        report.accept(severity);

        if (logOrThrowIfIncorrectPositionOrder) {
            logOrThrow(throwException, message);
        } else {
            LOGGER.warn(message);
        }

        return false;
```

Reply:
Yes, that's what the `logOrThrow` does. But in your case, you log it with the ERROR level, while it is usually logged on the WARN level.